### DISCUSSION TITLE
Setting up GRE decapsulation using local.zeek

### DISCUSSION THREAD

**Discussion by dudewhosurfs** [zeek/zeek#3945] [Help] [answered]
I am currently editing the local.zeek file to include a script to decapsulate GRE data.

Troubleshooting Process:
1. added base/protocols/tunnels to local.zeek (errored out)
2. created zeek.gre file to load into local.zeek (pictured below, errored out.)

The error is: "unknown identifier logging::Log::write, at or near log::write"

The zeek version is 3.1.5.

Any idea as to why this is failling? It has to do with the "logging::Log" line. I have made many changes to it but it still fails.

```
@load base/protocols/conn
@load base/frameworks/logging/main.zeek

module GRE;

export {
    redef enum Log::ID += { LOG };
    type Info: record {
        ts: time &log;
        uid: string &log;
        id: conn_id &log;
        proto: count &log;
        payload: string &log;
    };
}

event zeek_init() {
    Log::create_stream(GRE::LOG, [$columns=Info, $path="gre"]);
}

event gre_packet(c: connection, is_orig: bool, proto: count, payload: string) {
    local inner_packet = payload;
    logging::Log::write(GRE::LOG, [$ts=network_time(), $id=is_orig ? c$id$orig_h : c$id$resp_h, $proto=proto, $payload=inner_packet]);

    
    if ( proto == IPPROTO_IP || proto == IPPROTO_IPV6 ) {
        local analyzer = Analyzer::create("IP", c, is_orig);
        if ( analyzer != nil ) {
            analyzer$DeliverPacket(0, network_time(), payload);
        }
    }
}
```

**Comment by bbannier** [comment, marked as answer]
Hi,

We don't use GitHub discussions for support questions. Questions like this are best answered using on either slack or our forum which both are frequented by Zeek developers and professional users. You can join here:

Forum: [community.zeek.org](https://community.zeek.org/)

Slack: [zeek.org/community](https://zeek.org/community/)
